File: app/src/main/java/igrek/forceawaken/domain/task/LuckyTask.java
package igrek.forceawaken.domain.task;

import javax.inject.Inject;

import igrek.forceawaken.AwakenActivity;
import igrek.forceawaken.dagger.DaggerIOC;
import igrek.forceawaken.service.ui.info.UserInfoService;

public class LuckyTask implements AwakeTask {
	
	@Inject
	UserInfoService userInfoService;
	
	public LuckyTask() {
		DaggerIOC.getFactoryComponent().inject(this);
	}
	
	@Override
	public AwakeTask getInstance() {
		return new LuckyTask();
	}
	
	@Override
	public double getProbabilityWeight() {
		return 1;
	}
	
	@Override
	public void run(AwakenActivity activity) {
		userInfoService.showInfoBar("You are lucky today :)");
	}
}

File: app/src/main/java/igrek/forceawaken/task/LuckyTask.kt
package igrek.forceawaken.task

import igrek.forceawaken.activity.AwakenActivity
import igrek.forceawaken.info.UiInfoService
import igrek.forceawaken.inject.LazyExtractor
import igrek.forceawaken.inject.LazyInject
import igrek.forceawaken.inject.appFactory

class LuckyTask(
        uiInfoService: LazyInject<UiInfoService> = appFactory.uiInfoService,
) : AwakeTask {
    private val uiInfoService by LazyExtractor(uiInfoService)

    override val instance: AwakeTask
        get() = LuckyTask()
    override val probabilityWeight: Double
        get() = 1.0

    override fun run(activity: AwakenActivity) {
        uiInfoService.showSnackbar("You are lucky today :)")
    }
}